crates/re_types/src/lib.rs

```diff
@@ -256,8 +256,28 @@ pub trait AsComponents {
 /// Number of decimals shown for all vector display methods.
 pub const DISPLAY_PRECISION: usize = 3;
 
+/// Acrchetype are the high-level things you can log, like [`Image`][archetypes::Image], [`Points3D`][archetypes::Points3D], etc.
+///
+/// All archetypes implement the [`Archetype`] trait.
+///
+/// Each archetype is a collection of [`components`] and includes images, point clouds, and more.
```

Comment:
Each archetype is a collection of homogeneous batches of components (`ComponentBatch`).

I think that's a really important distinction that we must make clear to the user as soon as possible: it's all batches, do not think in terms of individual components ever.